service-node/src/main/scala/org/enterprisedlt/fabric/service/node/FabricNetworkManager.scala

```diff
@@ -391,19 +406,25 @@ class FabricNetworkManager(
     }
 
     //=========================================================================
-    def joinToChannel(channelName: String, joinRequest: JoinRequest): Either[String, Unit] = {
+    def joinToChannel(
+        channelName: String,
+        mspId: String,
+        caCerts: Array[String],
+        tlsCACerts: Array[String],
+        adminCerts: Array[String]
+    ): Either[String, Unit] = {
         getChannel(channelName)
           .map { channel =>
               val organizationDefinition = OrganizationDefinition(
-                  mspId = joinRequest.organization.mspId,
+                  mspId,
                   policies = PoliciesDefinition(
-                      admins = SignedByOneOf(MemberClassifier(joinRequest.organization.mspId, MSPRole.MSPRoleType.ADMIN)),
-                      writers = SignedByOneOf(MemberClassifier(joinRequest.organization.mspId, MSPRole.MSPRoleType.MEMBER)),
-                      readers = SignedByOneOf(MemberClassifier(joinRequest.organization.mspId, MSPRole.MSPRoleType.MEMBER))
+                      admins = SignedByOneOf(MemberClassifier(mspId, MSPRole.MSPRoleType.ADMIN)),
+                      writers = SignedByOneOf(MemberClassifier(mspId, MSPRole.MSPRoleType.MEMBER)),
+                      readers = SignedByOneOf(MemberClassifier(mspId, MSPRole.MSPRoleType.MEMBER))
                   ),
-                  caCerts = joinRequest.organizationCertificates.caCerts.map(Util.base64Decode).toSeq,
-                  tlsCACerts = joinRequest.organizationCertificates.tlsCACerts.map(Util.base64Decode).toSeq,
-                  adminCerts = joinRequest.organizationCertificates.adminCerts.map(Util.base64Decode).toSeq
+                  caCerts = caCerts.map(Util.base64Decode).toSeq,
```

Comment:
decoding should not be responsibility of this function - please move it outside i.e.:

`caCerts: Array[String]`
change to
`caCerts: Seq[Array[Byte]] `

and same for **tlsCACerts** **adminCerts**